packages/entity/src/rules/PrivacyPolicyRuleEnums.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Enum return type of privacy policy rule evaluation.
+ */
+export enum RuleEvaluationResult {
+  /**
+   * Deny viewer access to the entity.
+   */
+  DENY = -1,
+
+  /**
+   * Defer entity viewer access to subsequent rule in the privacy policy.
+   */
+  SKIP = 0,
+
+  /**
+   * Allow viewer access to the entity.
+   */
+  ALLOW = 1,
+}
+
+/**
+ * Estimated complexity of a privacy policy rule.
+ */
+export enum RuleComplexity {
+  /**
+   * Rule doesn't make any async calls.
+   */
+  CONSTANT_TIME = 0,
+
+  /**
+   * Rule makes an small amount of async calls. Definition of small dependent on application.
+   */
+  SMALL_ASYNC = 1,
+
+  /**
+   * Rule makes a large amount of async calls. Definition of large dependent on application.
+   */
+  LARGE_ASYNC = 2,
+}
```

Comment:
These enum value names mix several types of complexity:

- Async vs. sync
- I/O vs. no I/O
- Run time complexity - O(1), O(n), ...

An alternative way to categorize the complexity would be: NO_DATA_FETCHING, SMALL_DATA_FETCHING, LARGE_DATA_FETCHING.

Either way I think it would be useful to provide more guidance - even if it's just internally - about small vs. large.

For example: If you need to fetch an associated entity of an associated entity, that's just two entities but two round trips. If you need to fetch n associated entities, that's just one round trip. How do you compare these two access patterns?

Reply:
Okay, switching to just a number didn't go as well since there are default rules provided by the entity package that need a complexity rating.

Instead, I just made the categories LOW/MEDIUM/HIGH and left categorization and definition up to the application.